### PR TITLE
Fix Sqlite.err writing to parent dir

### LINKED ISSUE #8838: Sqite error file is writing to the parent of the output directory
Issue overview
--------------
The Sqlite error file is writing to the parent of the output directory, and the filename is munged with the output directory.

```
/parent/outDir/
/parent/outDirsqlite.err
```

This is happening here:

https://github.com/NREL/EnergyPlus/blob/82e97f1c21b8e7dd30a4278a93ad22bd284a7eb7/src/EnergyPlus/CommandLineInterface.cc#L488-L494

I think it should just be using ```composePath()``` like all of the other files since the suffix changes are handled above.

### Details
Some additional details for this issue (if relevant):
 - Platform (Mac, 10,15.7)
 - Version of EnergyPlus (V9.6.0 - 82e97f1c21)
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by mitchute**
Pull request overview
---------------------
 Fixes #8838 

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/CommandLineInterface.cc [hunk 1]
```diff
@@ -484,14 +484,7 @@ namespace CommandLineInterface {
...
         state.files.shade.filePath = composePath(shdSuffix + ".csv");
-
-        if (suffixType == "L") {
-            // out/sqlite.er:w
-            state.dataStrGlobals->outputSqliteErrFilePath = fs::path(state.dataStrGlobals->outDirPath.string() + sqliteSuffix + ".err");
-        } else {
-            // if 'D':  out/eplus-sqlite.err
-            state.dataStrGlobals->outputSqliteErrFilePath = composePath(sqliteSuffix + ".err");
-        }
+        state.dataStrGlobals->outputSqliteErrFilePath = composePath(sqliteSuffix + ".err");
```
Comment (jmarrec) [inline src/EnergyPlus/CommandLineInterface.cc]: Before filesystem changes:

https://github.com/NREL/EnergyPlus/blob/669aaac766488b51bc0c4fe58e5b1664ee4e61f1/src/EnergyPlus/CommandLineInterface.cc#L438-L442

Comment (jmarrec) [reply src/EnergyPlus/CommandLineInterface.cc]: I think the fix is incorrect as a result. It should be this I think no?

```suggestion
        if (suffixType == "L") {
            // out/sqlite.err
            state.dataStrGlobals->outputSqliteErrFilePath = state.dataStrGlobals->outDirPath / fs::path{sqliteSuffix + ".err"};
        } else {
            // if 'D':  out/eplus-sqlite.err
            state.dataStrGlobals->outputSqliteErrFilePath = composePath(sqliteSuffix + ".err");
        }
```

**Review by jmarrec — CHANGES_REQUESTED** [review]

File: src/EnergyPlus/CommandLineInterface.cc (hunk 1 above)
Comment (jmarrec) [reply src/EnergyPlus/CommandLineInterface.cc]: (of course, thank you for catching it!)

**Comment by Myoldmopar** [comment]
@mitchute do you agree with @jmarrec's suggested changes?  We can discuss further if there is still some uncertainty.

File: src/EnergyPlus/CommandLineInterface.cc (hunk 1 above)
Comment (mitchute) [reply src/EnergyPlus/CommandLineInterface.cc]: Yes, I agree with the suggested changes. I didn't catch the distinctions between the different suffix style types. I've tested these changes locally and get the following:

`-s L` -> "sqlite.err"
`-s C` -> "eplusSqlite.err"
`-s D` -> "eplus-sqlite.err"

Thanks, @jmarrec for the suggestion. I'll commit these changes and pull in develop one more time.

**Review by Myoldmopar — APPROVED** [review]

**Comment by Myoldmopar** [comment]
I'm not sure why Linux and Mac are lingering here, but this quick fix does not need to be held up.  Merging.  Thanks @mitchute and @jmarrec

**Pull request merged**